crates/trigger-http/src/server.rs

```diff
@@ -138,12 +148,43 @@ impl<F: RuntimeFactors> HttpServer<F> {
 
     /// Serve incoming requests over the provided [`TcpListener`].
     pub async fn serve(self: Arc<Self>) -> anyhow::Result<()> {
-        let listener = TcpListener::bind(self.listen_addr).await.with_context(|| {
-            format!(
-                "Unable to listen on {listen_addr}",
-                listen_addr = self.listen_addr
-            )
-        })?;
+        let listener: TcpListener = match TcpListener::bind(self.listen_addr).await {
+            Ok(listener) => listener,
+            Err(err) => {
+                if self.find_free_port && err.kind() == ErrorKind::AddrInUse {
+                    let mut found_listener = None;
+                    for _ in 1..=9 {
+                        let mut addr = self.listen_addr;
+                        addr.set_port(addr.port() + 1);
+
+                        match TcpListener::bind(addr).await {
+                            Ok(listener) => {
+                                found_listener = Some(listener);
+                                break;
+                            }
+                            Err(err) => {
+                                if err.kind() == ErrorKind::AddrInUse {
+                                    continue;
+                                }
+                                return Err(anyhow::anyhow!("Unable to listen on {}", addr));
```

Comment:
Is it correct that we bail here?  Why would an error on port 3002 stop us from trying port 3003?  (Is it, perhaps, that errors other than AddrInUse imply we've gone to a bad place?)

Reply:
I understand that it continues on `AddrInUse`.  I am asking why it _bails_ on other errors.  Is the assumption that other errors are unrecoverable and there is no point searching further?

Reply:
If it's going to turn into a faff then don't worry about it - stick with what you have and we can always re-evaluate if something comes up (i.e. no need to raise an issue either).